lib/compiler/src/compile.erl

```diff
@@ -1726,6 +1728,15 @@ native_compile_1(Code, St) ->
 	    end
     end.
 
+%% We have an optional runtime dependency on HiPE, but there's no way to
+%% suppress warnings for calls to unknown functions when -Wunknown is on, so
+%% we'll run all HiPE calls through here to suppress warnings when HiPE hasn't
+%% been compiled. :(
+dialyzer_whining_inhibitor(M, F, A) ->
+  apply(id(M), id(F), id(A)).
```

Comment:
that's a fun one. Would it have made sense to use `-dialyzer({nowarn_function, Name/Arity}).` for some of that stuff?

Reply:
That's what we tried at first, but it won't suppress warnings for calls to unknown functions. :/